webapp/components/survey/CategorySelector/CategorySelector.js

```diff
@@ -13,29 +12,20 @@ import { useSurvey } from '@webapp/store/survey'
 import Dropdown from '@webapp/components/form/Dropdown'
 import PanelRight from '@webapp/components/PanelRight'
 import CategoryList from '@webapp/views/App/views/Designer/CategoryList'
-import CategoryView from '@webapp/loggedin/surveyViews/category/categoryView'
-
-import * as CategoryState from '@webapp/loggedin/surveyViews/category/categoryState'
-import * as CategoryActions from '@webapp/loggedin/surveyViews/category/actions'
+import CategoryDetails from '@webapp/components/survey/CategoryDetails'
 
 const CategorySelector = (props) => {
   const { disabled, categoryUuid, validation, showManage, showAdd, onChange } = props
 
   const i18n = useI18n()
 
-  const dispatch = useDispatch()
-
   const [showCategoriesPanel, setShowCategoriesPanel] = useState(false)
+  const [showCategoryPanel, setShowCategoryPanel] = useState(false)
+  const [categoryUuidToEdit, setCategoryUuidToEdit] = useState(null)
```

Comment:
don't you show category panel when the categoryUuidToEdit is set? you might need only one of the two?

Reply:
right, but when you add a new category you just set showCategoryPanel to true and the category will be created by the init action, you don't know the uuid of the newly created category at that moment

Reply:
when you create a category, the server responds with the category. in that moment you do have the category uuid. or am I missing something?

Reply:
actually the category is created in the client and inserted through the API.
Now this is done from the CategoryDetails / init action when the component is mounted an categoryUuid is not passed to it.
The same thing happens for the taxonomy.
What should we do? Call a CategoryDetails action from CategorySelector and from CategoryList to create new categories?